lib/teiserver/battle/libs/balance_lib.ex

```diff
@@ -130,24 +162,43 @@ defmodule Teiserver.Battle.BalanceLib do
     )
   end
 
+  # Only take keys we need
+  defp clean_groups(groups) do
+    groups |> Enum.map(fn x->
+      Map.take(x, ~w(members count group_rating ratings)a)
+    end)
+  end
+
   # Take the balance result and add some extra fields to make using it easier
   defp expand_balance_result(balance_result) do
     team_groups =
-      balance_result.teams
-      |> Map.new(fn {team_id, groups} ->
-        {team_id, Enum.reverse(groups)}
-      end)
+      cond do
+        Map.has_key?(balance_result, :team_groups) ->
+          balance_result.team_groups
+
+        true ->
+          balance_result.teams
+          |> Map.new(fn {team_id, groups} ->
+            {team_id, Enum.reverse(clean_groups((groups)))}
+          end)
```

Comment:
A more idiomatic way of checking that is to use pattern matching:
```elixir
case balance_result[:team_groups] do
  groups -> groups
  nil -> # do stuff
end
```

Reply:
Noted. Is there a difference between 
```
balance_result[:team_groups]
```
and
```
balance_result.team_groups
```
Can I use either?